### ISSUE TITLE
commit 50f1f1dc breaks ctrl-l (clear screen)

### ISSUE THREAD

**Issue by octoploid** [tmux/tmux#924]
commit 50f1f1dc breaks ctrl-l (clear screen)
Instead tmux just jumps to first line without clearing the screen, 
overwriting the existing content.

**Comment by nicm** [comment]
This issue does not include enough information. Please read the [CONTRIBUTING](https://raw.githubusercontent.com/tmux/tmux/master/CONTRIBUTING) file.

**Comment by octoploid** [comment]
Well: type "dmesg" then hit ctrl-l. The cursor jumps to line 1 in the terminal without clearing the screen.

**Comment by nicm** [comment]
Right, but if it didn't work for me, then I wouldn't need any more information, would I?

**Comment by octoploid** [comment]
./tmux -vvvv -Ltest -f /dev/null new
[log.tar.zip](https://github.com/tmux/tmux/files/998422/log.tar.zip)

**Comment by nicm** [comment]
Hmm, looks OK. What terminal are you using?

**Comment by nicm** [comment]
Oh I bet this is the Konsole thing again, let me try that.

**Comment by octoploid** [comment]
Right: konsole

**Comment by nicm** [comment]
Yeah, we are clearing too many lines. This should fix it:
~~~~
diff --git a/tty.c b/tty.c
index d638f7d0..89f19101 100644
--- a/tty.c
+++ b/tty.c
@@ -838,7 +838,7 @@ tty_clear_area(struct tty *tty, const struct window_pane *wp, u_int py,
                    tty_term_has(tty->term, TTYC_INDN)) {
                        tty_region(tty, py, py + ny - 1);
                        tty_margin_off(tty);
-                       tty_putcode1(tty, TTYC_INDN, ny);
+                       tty_putcode1(tty, TTYC_INDN, ny - 1);
                        return;
                }

**Comment by octoploid** [comment]
Yes, thanks for the fix.

**Comment by nicm** [comment]
Great, thanks!

**Comment by thejmazz** [comment]
Fix worked for me as well using Konsole 17.04.0 and tmux `9a1b9f1` (and it still works in gnome-terminal too).

**Comment by CoffeeFrame** [comment]
I am having this same problem with qterminal & Cool Retro Term. No problem with gnome terminal or headless system.

**Comment by schirmeier** [comment]
If anyone wonders: The mentioned fix (7a4c66b) was reverted two days later (cb5fcb3): "The Konsole SU bug means it can't clear the entire scroll region (it ignores if >= size, not if > as I first thought). So we can't effectively fix it in code - remove the workarounds which just cause bugs on other terminals."

I still see this bug in Ubuntu 17.10 (Konsole 17.04.3-0ubuntu1) with tmux 2.6.
